### PR TITLE
Dock widgets with no unnecessary title bars under Windows.

### PR DESCRIPTION
**Pull request by jacek-izykowski**

### CODE REVIEW

File: src/ui/veles_mainwindow.cc [hunk 1]
```diff
@@ -413,6 +590,66 @@ DockWidget* MainWindowWithDetachableDockWidgets::findDockNotTabifiedWith(
...
+
+    auto main_window = dynamic_cast<MainWindowWithDetachableDockWidgets*>(obj);
+    if(main_window) {
+      return main_window;
+    }
+
+    obj = obj->parent();
+  }
+
+  return nullptr;
```
Comment (adamiwaniuk) [inline src/ui/veles_mainwindow.cc]: czy ten kod jest osiagalny ?, moze lepiej zamienic te true w petli na obj != nullptr

Comment (jacek-izykowski) [inline src/ui/veles_mainwindow.cc]: I do agree. This additional return statement was meant to avoid compiler warnings.

**Comment by adamiwaniuk** [comment]
This dosn't work well, for example: open fisrt file and open visualization (title bar is presnet)

**Comment by jacek-izykowski** [comment]
@adamiwaniuk Fixed.

File: src/ui/veles_mainwindow.cc [hunk 2]
```diff
@@ -413,6 +590,66 @@ DockWidget* MainWindowWithDetachableDockWidgets::findDockNotTabifiedWith(
...
+    }
+  }
+
+  return nullptr;
+}
+
+MainWindowWithDetachableDockWidgets* MainWindowWithDetachableDockWidgets
+    ::getParentMainWindow(QObject* obj) {
+  while (obj != nullptr) {
+    if(obj == nullptr) {
```
Comment (adamiwaniuk) [inline src/ui/veles_mainwindow.cc]: this condition is always false, we don't need this "if"

Comment (jacek-izykowski) [inline src/ui/veles_mainwindow.cc]: Poprawione.

File: src/ui/veles_mainwindow.cc [hunk 3]
```diff
@@ -157,6 +161,32 @@ void DockWidget::detachToNewTopLevelWindowAndMaximize() {
...
+
+  if (parent) {
+    parent->updateDocksAndTabs();
+  }
+}
+
+void DockWidget::switchTitleBar(bool is_default) {
+  if (is_default) {
+    if (titleBarWidget() != nullptr) {
+      setTitleBarWidget(0);
```
Comment (MaciekPytel) [inline src/ui/veles_mainwindow.cc]: nitpicking - why use 0 instead of nullptr?

Comment (jacek-izykowski) [inline src/ui/veles_mainwindow.cc]: Poprawione,

File: src/ui/veles_mainwindow.cc [hunk 4]
```diff
@@ -413,6 +590,62 @@ DockWidget* MainWindowWithDetachableDockWidgets::findDockNotTabifiedWith(
...
+    // QDockWidgets (stored as quintptr) and can be retrieved through public
+    // interface.
+    QDockWidget* dock =
+        reinterpret_cast<QDockWidget*>(
+        qvariant_cast<quintptr>(tab_bar->tabData(index)));
+
+    return dock;
+  }
+
+  return 0;
```
Comment (MaciekPytel) [inline src/ui/veles_mainwindow.cc]: nullptr

Comment (jacek-izykowski) [inline src/ui/veles_mainwindow.cc]: Fixed.

**Comment by MaciekPytel** [comment]
Rebase before merge please

**Pull request merged**